nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/vision/AbstractStartGcpVisionOperation.java

```diff
@@ -70,7 +92,15 @@ public void onStopped() throws IOException {
     }
 
     private InputStream getInputStreamFromProperty(ProcessContext context, FlowFile flowFile) {
-        return new ByteArrayInputStream(context.getProperty(getJsonPayloadPropertyDescriptor()).evaluateAttributeExpressions(flowFile).getValue().getBytes(StandardCharsets.UTF_8));
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(OUTPUT_BUCKET.getName(), getAttributeValue(context, flowFile, OUTPUT_BUCKET.getName()));
+        attributes.put(FEATURE_TYPE.getName(), getAttributeValue(context, flowFile, FEATURE_TYPE.getName()));
+        return new ByteArrayInputStream(context.getProperty(getJsonPayloadPropertyDescriptor())
+                .evaluateAttributeExpressions(flowFile, attributes).getValue().getBytes(StandardCharsets.UTF_8));
+    }
+
+    private String getAttributeValue(ProcessContext context, FlowFile flowFile, String name) {
+        return flowFile.getAttribute(name) != null ? flowFile.getAttribute(name) : context.getProperty(name).evaluateAttributeExpressions().getValue();
```

Comment:
It would be helpful to split this into multiple lines for improved readability, and adjust the logical operator. The `evaluateAttributeExpressions()` invocation does not add much value in this case, so it seems unnecessary.
```suggestion
        final String flowFileAttribute = flowFile.getAttribute(name);
        final PropertyValue propertyValue = context.getProperty(name);
        return flowFileAttribute == null ? propertyValue.getValue() : flowFileAttribute;
```